### PR TITLE
Fix issue with below limit strings in ngram calculation

### PR DESCRIPTION
**Pull request by Vortexx2**
When strings provided were below `n` characters, the ngram function returns an empty list. This was previously exploded, without filtering out the empty lists, causing the <NA> token to occur erroneously. Now, the empty lists should be filtered out.

## Description
Changes Proposed:
Should close the issue [BUG] #14684.
Addressed an issue in the ngram function where input strings below a specified length (n) resulted in the function returning an empty list. Previously, this empty list was not filtered out during further processing, leading to the inadvertent occurrence of the <NA> token.

Solution:
Modified the code to include a filtering step that removes empty lists generated by the ngram function when input strings are below the specified length. This ensures that the <NA> token is no longer erroneously introduced.

Impact:
This enhancement improves the accuracy and reliability of the ngram function by handling edge cases where input strings are shorter than the specified length. The filtering of empty lists prevents unintended consequences, specifically the occurrence of the <NA> token in such scenarios.

## Checklist
- [✅ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ❌ ] New or existing tests cover these changes.
- [✅ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/core/column/string.py [hunk 1]
```diff
@@ -4843,6 +4843,7 @@ def character_ngrams(
         result = self._return_or_inplace(lc, retain_index=True)
 
         if isinstance(result, cudf.Series) and not as_list:
+            result = result[result.list.len() > 0] # before exploding, removes those lists which have 0 length
```
Comment (davidwendt) [inline python/cudf/cudf/core/column/string.py:4846]: Seems this fix could go in the calling code (cuml?). 
Just pass `as_list=True` and do the filter step and explode outside this function.

**Review by davidwendt — CHANGES_REQUESTED** [review]
Just wondering if this fix is an appropriate, breaking change to cuDF behavior.

File: python/cudf/cudf/core/column/string.py (hunk 1 above)
Comment (Vortexx2) [reply python/cudf/cudf/core/column/string.py:4846]: This fix should be included in the cuDF repo due to it being a bug in the `character_ngrams` function itself. sklearn does not have this underlying issue with the `sklearn.feature_extraction.text.CountVectorizer` class that causes NA tokens to be returned on empty strings for example.

Comment (davidwendt) [reply python/cudf/cudf/core/column/string.py:4846]: Ok. Could you add a pytest for this?

Comment (Vortexx2) [reply python/cudf/cudf/core/column/string.py:4846]: Sure

Comment (davidwendt) [reply python/cudf/cudf/core/column/string.py:4846]: Any progress on this @Vortexx2 ?

**Comment by davidwendt** [comment]
Closing this in favor of #15371
